### PR TITLE
Resizing the game window now works. Kind of. A bit.

### PR DESCRIPTION
**Pull request by Adrijaned**
# TO DO
 - UI still not resizes correctly - but neither did it before, and its mostly actually placeholders now (https://github.com/MovingBlocks/DestinationSol/blob/develop/engine/src/main/java/org/destinationsol/menu/MenuLayout.java)
 - Mouse is still a bit off.
 - lots of other things are definitely broken too, now I'm just glad it compiles, more work may follow in this/next PRs

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/destinationsol/CommonDrawer.java [hunk 1]
```diff
@@ -18,45 +18,58 @@
...
 import org.destinationsol.ui.UiDrawer;
 
 public class CommonDrawer {
-    public final float width;
-    public final float height;
-    public final float dimensionsRatio;
+    public DimensionsRatio dimensionsRatio;
+    public float width;
+    public float height;
+    public float r;
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/CommonDrawer.java]: radius? record? resolution? root? ransomware? rhinoceros? could even be ratio but who really knows

File: engine/src/main/java/org/destinationsol/CommonDrawer.java [hunk 2]
```diff
@@ -18,45 +18,58 @@
...
+        r = width / height;
         spriteBatch = new SpriteBatch();
 
         font = Assets.getFont("engine:main").getBitmapFont();
 
         originalFontHeight = font.getXHeight();
 
         layout = new GlyphLayout();
+
+        orthographicCamera = new OrthographicCamera(1024, 768);
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/CommonDrawer.java]: could/should we pull the resolution out of settings initially instead?

File: engine/src/main/java/org/destinationsol/SolApplication.java [hunk 3]
```diff
@@ -28,19 +30,14 @@
...
 import org.destinationsol.game.context.internal.ContextImpl;
-import org.destinationsol.assets.audio.OggMusicManager;
-import org.destinationsol.assets.audio.OggSoundManager;
 import org.destinationsol.menu.MenuScreens;
-import org.destinationsol.ui.DebugCollector;
-import org.destinationsol.ui.FontSize;
-import org.destinationsol.ui.SolInputManager;
-import org.destinationsol.ui.SolLayouts;
-import org.destinationsol.ui.UiDrawer;
+import org.destinationsol.ui.*;
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/SolApplication.java]: star

File: engine/src/main/java/org/destinationsol/game/SolCam.java [hunk 4]
```diff
@@ -39,17 +41,21 @@
...
 
     private float myPrevHeroLife;
     private float myShake;
     private float myAngle;
     private float myZoom;
     private Vector2 position;
+    private DimensionsRatio dimensionsRatio;
 
-    public SolCam(float r) {
+    public SolCam(DimensionsRatio r) {
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/game/SolCam.java]: r here too

File: engine/src/main/java/org/destinationsol/game/SolGame.java [hunk 5]
```diff
@@ -31,11 +33,7 @@
 import org.destinationsol.game.drawables.DrawableDebugger;
 import org.destinationsol.game.drawables.DrawableManager;
 import org.destinationsol.game.farBg.FarBackgroundManagerOld;
-import org.destinationsol.game.item.ItemContainer;
-import org.destinationsol.game.item.ItemManager;
-import org.destinationsol.game.item.LootBuilder;
-import org.destinationsol.game.item.MercItem;
-import org.destinationsol.game.item.SolItem;
+import org.destinationsol.game.item.*;
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/game/SolGame.java]: star

File: engine/src/main/java/org/destinationsol/game/drawables/DrawableManager.java [hunk 6]
```diff
@@ -23,19 +23,9 @@
...
-import org.destinationsol.game.SolObject;
-
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import org.destinationsol.game.*;
+
+import java.util.*;
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/game/drawables/DrawableManager.java]: double star

File: engine/src/main/java/org/destinationsol/ui/DimensionsRatio.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+ */
+package org.destinationsol.ui;
+
+
+/**
+ * Should be equal to {@code ((float) gameWindowWidth) / ((float) gameWindowHeight)}.
+ * <p>
+ * Old {@code r} variable in lots of places.
+ */
+public class DimensionsRatio {
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/ui/DimensionsRatio.java]: just wondering, why do we need a custom class for a float value?

**Comment by NicholasBatesNZ** [comment]
Actually, after some brief testing, I discovered a crash when firing up the ships engine that is not present in develop

```
com.badlogic.gdx.utils.GdxRuntimeException: java.lang.AssertionError: Attempted to loop a sound without a parent object: core:medEngineWork2
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication$1.run(LwjglApplication.java:127)
Caused by: java.lang.AssertionError: Attempted to loop a sound without a parent object: core:medEngineWork2
	at org.destinationsol.game.sound.OggSoundManager.play(OggSoundManager.java:164)
	at org.destinationsol.game.sound.OggSoundManager.play(OggSoundManager.java:137)
	at org.destinationsol.game.particle.DSParticleEmitter.setWorking(DSParticleEmitter.java:237)
	at org.destinationsol.game.particle.PartMan.updateAllHullEmittersOfType(PartMan.java:104)
	at org.destinationsol.game.ship.ShipEngine.update(ShipEngine.java:43)
	at org.destinationsol.game.ship.hulls.Hull.update(Hull.java:113)
	at org.destinationsol.game.ship.SolShip.update(SolShip.java:228)
	at org.destinationsol.game.ObjectManager.update(ObjectManager.java:94)
	at org.destinationsol.game.SolGame.update(SolGame.java:277)
	at org.destinationsol.SolApplication.update(SolApplication.java:158)
	at org.destinationsol.SolApplication.safeUpdate(SolApplication.java:133)
	at org.destinationsol.SolApplication.render(SolApplication.java:112)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication.mainLoop(LwjglApplication.java:215)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication$1.run(LwjglApplication.java:120)
```

**Comment by Adrijaned** [comment]
Merged with #321

**Pull request closed without merging**